store/imagestore/remote.go

```diff
@@ -19,9 +19,14 @@ package imagestore
 
 import (
 	"database/sql"
+	"errors"
 	"time"
 )
 
+var (
+	ErrRemoteNotFound = errors.New("remote with the given ACI URL not found")
```

Comment:
Just a personal taste, but I find easier to just return nil if not found end check it in the caller instead of the error type checking. Many APIs (with the exception of os that wraps a syscall error for ErrNotFound) do this.